### PR TITLE
[WIP] Implement IgniteEntityEvent

### PR DESCRIPTION
**Pull request by Maxqia**

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 1]
```diff
@@ -161,7 +167,7 @@
     private net.minecraft.entity.Entity teleportVehicle;
     private float origWidth;
     private float origHeight;
-    @Nullable private DamageSource originalLava;
+    @Nullable private DamageSource originalLava, originalFire, originalLightning;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: Make these individual fields. Don't take syntactical shortcuts like this.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 2]
```diff
@@ -147,8 +152,9 @@
 public abstract class MixinEntity implements IMixinEntity {
 
     private static final String LAVA_DAMAGESOURCE_FIELD = "Lnet/minecraft/util/DamageSource;LAVA:Lnet/minecraft/util/DamageSource;";
-    private static final String ATTACK_ENTITY_FROM_METHOD = "Lnet/minecraft/entity/Entity;attackEntityFrom(Lnet/minecraft/util/DamageSource;F)Z";
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:150]: Line noise. don't move static fields around in this case.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 3]
```diff
@@ -240,6 +245,42 @@
 
     // @formatter:on
 
+    /**
+     * Overwritten due to pass-by-value nature of Java
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: Explains nothing to anyone. Why was this overwritten.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 4]
```diff
@@ -240,6 +245,42 @@
...
+    @Overwrite
+    public void setFire(int seconds) {
+        int ticks = seconds * 20;
+
+        if (((Object)this) instanceof EntityLivingBase) {
+            ticks = EnchantmentProtection.getFireTimeForEntity((EntityLivingBase) (Object) this, ticks);
+        }
+
+        if (!world.isRemote) {
+            PhaseContext context = ((IMixinWorldServer) world).getCauseTracker().getCurrentContext();
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:261]: What are the comments for saying "Here is where sponge is making a change.... This other part is vanilla code."

Look at other examples of overwrites (MixinWorld or MixinChunk even).

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 5]
```diff
@@ -240,6 +245,42 @@
...
+
+        if (this.fire < ticks) {
+            this.fire = ticks;
+        }
+    }
+
+    @Inject(method = "attackEntityFrom", at=@At("HEAD"))
+    public void onAttackEntityFrom(DamageSource source, float amount, CallbackInfoReturnable<Boolean> cir) {
+        if (!world.isRemote) {
+            ((IMixinWorldServer) world).getCauseTracker().getCurrentContext().add(NamedCause.of(NamedCause.SOURCE, source));
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:282]: Why?

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 6]
```diff
@@ -360,6 +401,25 @@ public void postDealFireDamage(CallbackInfo callbackInfo) {
         }
     }
 
+    @Inject(method = "onStruckByLightning", at = @At(value = "FIELD", target = LIGHTNING_DAMAGESOURCE_FIELD, opcode = Opcodes.GETSTATIC))
+    public void preLightning(EntityLightningBolt lightningBolt, CallbackInfo callbackInfo) {
+        if (!this.world.isRemote) {
+            this.originalLightning = DamageSource.LIGHTNING_BOLT;
+            DamageSource.IN_FIRE = new EntityDamageSource("lightningBolt", lightningBolt);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: Uh, what?

**Review by gabizou — CHANGES_REQUESTED** [review]
A little sloppy. You provide no real explanation as to what you're aiming to do here. PR has no description.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java [hunk 7]
```diff
@@ -360,6 +401,25 @@ public void postDealFireDamage(CallbackInfo callbackInfo) {
...
+        }
+    }
+
+    @Inject(method = "onStruckByLightning", at = @At(value = "INVOKE_ASSIGN", target = ATTACK_ENTITY_FROM_METHOD))
+    public void postLightning(EntityLightningBolt lightningBolt, CallbackInfo callbackInfo) {
+        if (!this.world.isRemote) {
+            if (this.originalLightning == null) {
+                SpongeImpl.getLogger().error("Original lightning is null!");
+                Thread.dumpStack();
```
Comment (liach) [inline src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:419]: Is this test code?

Comment (Maxqia) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:419]: This was mostly inspired by postDealFireDamage/postSetOnFire, which includes that for some reason.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java (hunk 5 above)
Comment (Maxqia) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:282]: Needed context for setFire and this is always called before it.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java (hunk 6 above)
Comment (Maxqia) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: Mainly did this to pass the entity (the lightning bolt) through as a source, which has a representation in Sponge.

Comment (liach) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: Seems you never set this back.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java (hunk 7 above)
Comment (liach) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java:419]: You can probably just make a throwable, I think.

File: src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java (hunk 6 above)
Comment (Maxqia) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: 0-0 wait wat, that's the wrong field!

Comment (liach) [reply src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java]: 😆

**Comment by parlough** [comment]
@Maxqia Do you plan to update this?

**Comment by liach** [comment]
I don't think he's active.

**Comment by gabizou** [comment]
Pending 14 days from today, this PR is going to be closed for being stale, no development has been made.

**Comment by parlough** [comment]
Going to close this due to inactivity, thanks for your contribution though!

Has also been superseded by https://github.com/SpongePowered/SpongeCommon/pull/1708

**Pull request closed without merging**